go/test/endtoend/backup/vtbackup/backup_only_test.go

```diff
@@ -0,0 +1,304 @@
+	// Test First Backup flow.
+	//
+	//    firstBackupTest will:
+	//    - create a shard with master and replica1 only
+	//    - run InitShardMaster
+	//    - insert some data
+	//    - take a backup
+	//    - insert more data on the master
+	//    - bring up replica2 after the fact, let it restore the backup
+	//    - check all data is right (before+after backup data)
+	//    - list the backup, remove it
+
+	// Store initial backup counts
+	backupsCount := countBackups(t)
+
+	// insert data on master, wait for slave to get it
+	_, err := master.VttabletProcess.QueryTablet(vtInsertTest, keyspaceName, true)
+	assert.Nil(t, err)
+	// Add a single row with value 'test1' to the master tablet
+	_, err = master.VttabletProcess.QueryTablet("insert into vt_insert_test (msg) values ('test1')", keyspaceName, true)
+	assert.Nil(t, err)
+
+	// Check that the specified tablet has the expected number of rows
+	cluster.VerifyRowsInTablet(t, replica1, keyspaceName, 1)
+
+	// backup the slave
+	log.Info("taking backup %s", time.Now())
+	vtBackup(t, false)
+	log.Info("done taking backup %s", time.Now())
+
+	// check that the backup shows up in the listing
+	backups := countBackups(t)
+	assert.Equal(t, backups, backupsCount+1)
+
+	// insert more data on the master
+	_, err = master.VttabletProcess.QueryTablet("insert into vt_insert_test (msg) values ('test2')", keyspaceName, true)
+	assert.Nil(t, err)
+	cluster.VerifyRowsInTablet(t, replica1, keyspaceName, 2)
+
+	// now bring up the other slave, letting it restore from backup.
+	err = localCluster.VtctlclientProcess.InitTablet(replica2, cell, keyspaceName, hostname, shardName)
+	assert.Nil(t, err)
+	restore(t, replica2, "replica", "SERVING")
+	// Replica2 takes time to serve. Sleeping for 5 sec.
+	time.Sleep(5 * time.Second)
+	//check the new slave has the data
+	cluster.VerifyRowsInTablet(t, replica2, keyspaceName, 2)
+
+	// check that the restored slave has the right local_metadata
+	result, err := replica2.VttabletProcess.QueryTabletWithDB("select * from local_metadata", "_vt")
+	assert.Nil(t, err)
+	assert.Equal(t, replica2.Alias, result.Rows[0][1].ToString(), "Alias")
+	assert.Equal(t, "ks.0", result.Rows[1][1].ToString(), "ClusterAlias")
+	assert.Equal(t, cell, result.Rows[2][1].ToString(), "DataCenter")
+	if tabletType == "replica" {
+		assert.Equal(t, "neutral", result.Rows[3][1].ToString(), "PromotionRule")
+	} else {
+		assert.Equal(t, "must_not", result.Rows[3][1].ToString(), "PromotionRule")
+	}
+
+	removeBackups(t)
+	backups = countBackups(t)
+	assert.Equal(t, 0, backups)
+
+}
+
+func vtBackup(t *testing.T, initialBackup bool) {
+	// Take the back using vtbackup executable
+	extraArgs := []string{"-allow_first_backup", "-db-credentials-file", dbCredentialFile}
+	log.Info("starting backup tablet %s", time.Now())
+	err := localCluster.StartVtbackup(newInitDBFile, initialBackup, keyspaceName, shardName, cell, extraArgs...)
+	assert.Nil(t, err)
+}
+
+func listBackups(t *testing.T) string {
```

Comment:
can these local functions be replaced by shared code?

Reply:
https://github.com/vitessio/vitess/blob/master/go/test/endtoend/cluster/cluster_util.go#L77

Reply:
Yes, that was the plan. 
But Pradip also started making changes in that common code. And if I update it in my PR then there will be a lot of merge conflicts. We will make sure it is cleaned up in his PR.

Reply:
ok I'll approve for now and you can fix this in the other PR.